### PR TITLE
Update provider flags according to project ones

### LINKED ISSUE #49463: "evaluate default value on providers side" is not saved as checked when saving projects on QGIS 3.26
### What is the bug or the crash?

I have a Postgis table whose id is generated by a sequence.
Previously, on GIS 3.24 an older versions, when I inserted it without saving, it used to write down the id, but in 3.26, it writes the sequence and the id is retrieved after the object is saved. It causes problem when there are other layers joined with this object using its id. I should first save and then join other objects to it.

 

### Versions

QGIS version
3.26.1-Buenos Aires
QGIS code revision
b609df9ed4d
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.5.1
PROJ version
9.0.1
EPSG Registry database version
v10.064 (2022-05-19)
GEOS version
3.10.3-CAPI-1.16.1
SQLite version
3.38.1
PDAL version
2.3.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows Server 2016 Version 1607

Active Python plugins
logs_searcher
1.0.5
networkanalysis
1.9.1
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
sagaprovider
2.12.99

-----
QGIS version
3.24.3-Tisler
QGIS code revision
cf22b74e
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.4.3
PROJ version
9.0.0
EPSG Registry database version
v10.054 (2022-02-13)
GEOS version
3.10.2-CAPI-1.16.0
SQLite version
3.38.1
PDAL version
2.3.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows Server 2016 Version 1607

Active Python plugins
logs_searcher
1.0.5
networkanalysis
1.9.1
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
sagaprovider
2.12.99
------

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by troopa81**
Fixes #49463

Since e4db7faed7339ce46d5c6f29077dc27eaeebad89 we need to call *setFlags()* to update providers when we read flags from project

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]
Could you add a test for this too?

**Comment by troopa81** [comment]
Done

File: tests/src/python/test_qgsproject.py [hunk 1]
```diff
@@ -1566,6 +1567,36 @@ def test_remember_editable_status(self):
         self.assertFalse(project4.mapLayer(layer_b.id()).isEditable())
         self.assertFalse(project4.mapLayer(layer_c.id()).isEditable())
 
+    def test_remember_evalutate_defaut_values(self):
```
Comment (nyalldawson) [inline tests/src/python/test_qgsproject.py:1570]:
```suggestion
    def test_remember_evaluate_default_values(self):
```

File: tests/src/python/test_qgsproject.py [hunk 2]
```diff
@@ -1566,6 +1567,36 @@ def test_remember_editable_status(self):
...
+        """
+        Test that EvaluateDefaultValues property is correctly set when loading project
+        """
+
+        project = QgsProject()
+        project.removeAllMapLayers()
```
Comment (nyalldawson) [inline tests/src/python/test_qgsproject.py:1576]:
```suggestion
```

**Pull request merged**